pkg/authentication/firebase.go

```diff
@@ -48,3 +50,31 @@ func NewFirebaseWithTokenVerifier(firebaseAuth FirebaseTokenVerifier) Authentica
 		firebaseAuth: firebaseAuth,
 	}
 }
+
+// firebaseAuth uses the firebase application to refresh the keys used to verify the token signature.
+// The firebase.App has an internal mechanism to avoid repeating this operation for every request.
+type firebaseAuth struct {
+	client *auth.Client
+}
+
+// VerifyIDToken gets a new public key in case a key rotation has been requested, and verifies the given token.
+func (auth *firebaseAuth) VerifyIDToken(ctx context.Context, idToken string) (*auth.Token, error) {
+	return auth.client.VerifyIDToken(ctx, idToken)
+}
+
+// NewFirebase initializes a new FirebaseTokenVerifier implementation using a Firebase application, and it's in
+// charge of refreshing the public key used to verify tokens every time a new key rotation happens.
+//
+//	auth := NewFirebaseWithTokenVerifier(NewFirebase(app))
+//	if err := auth.VerifyJWT(ctx, token); err != nil {
+//		log.Fatalf("failed to verify jwt: %v\n", err)
+//	}
+func NewFirebase(app *firebase.App) FirebaseTokenVerifier {
+	client, err := app.Auth(context.Background())
+	if err != nil {
+		log.Fatalf("failed to initialize Firebase client: %v\n", err)
```

Comment:
`log.Fatalf` calls `os.Exit(1)` after logging. 

Let's leave the error handling up to the applications that use this client and return an error instead.

Reply:
I'd prefer if we don't return errors when initializing implementations. Why? 

`auth := NewFirebaseWithTokenVerifier(NewFirebase(app))`

I think developers will appreciate having an easy way to set up authentication like the example included in the docs.

**We can either:**
- Revert the changes to initializing the client in the `VerifyIDToken` and add an if statement if the `*auth.Client` is different than `nil`.
- Keep the `log.Fatalf`.

Reply:
> Revert the changes to initializing the client in the VerifyIDToken and add an if statement if the *auth.Client is different than nil.

This will require adding this `if` to all methods that we add to this, adding an extra check that I don't know if it makes sense.

A third alternative is to return an `error` and have the auth initialization be a bit more verbose. Personally, I would do the following with the current configuration

```go
firebaseAuth := NewFirebase(app)
auth := NewFirebaseWithTokenVerifier(firebaseAuth)
```

Returning an error would change it to

```go
firebaseAuth, err := NewFirebase(app)
if err != nil {
  // Handle error
}
auth := NewFirebaseWithTokenVerifier(firebaseAuth)
```

which I don't find too bad.

Applications that don't strictly need Firebase to function will not be able to use this auth implementation with the current setup.

Reply:
> This will require adding this if to all methods that we add to this, adding an extra check that I don't know if it makes sense.

I don't think I agree. We don't even know if we'll ever add more methods, and if we ever need to, we can add a `client` internal method that does it for you, extracting the if statement to a single place.

> which I don't find too bad.

I do find it a bit annoying to use, it makes things harder for the developer to set up auth when wanting to follow a more functional approach:

```go
BearerToken(NewFirebaseWithTokenVerifier(NewFirebase(app)))
```

Reply:
This is the exact discussion we had in our previous PR, only each of us is on the other side now. To me, having three (long) constructors chained makes it difficult to read and follow, and not having the option to react on Firebase.

The truth is that this is not relevant for the current intended use cases.

I have an alternative proposal: let's have two instantiation functions: `NewFirebaseOrExit` that exits if it fails to initialize and `NewFirebase` that doesn't. That way we can support both use cases and not have to call an internal method each time we call any other method.

But let's please not go with adding a `Client` method. This extra step per call can be completely avoided with a different approach.

Reply:
Ok, alright, I'll use the original proposal, the one where

```go
firebaseAuth, err := NewFirebase(app)
if err != nil {
  // Handle error
}
```